kernel/src/expressions/scalars.rs

```diff
@@ -254,10 +261,21 @@ impl PartialOrd for Scalar {
             (Date(_), _) => None,
             (Binary(a), Binary(b)) => a.partial_cmp(b),
             (Binary(_), _) => None,
-            (Decimal(_, _, _), _) => None, // TODO: Support Decimal
-            (Null(_), _) => None,          // NOTE: NULL values are incomparable by definition
-            (Struct(_), _) => None,        // TODO: Support Struct?
-            (Array(_), _) => None,         // TODO: Support Array?
+            (Decimal(v1, p1, s1), Decimal(v2, p2, s2)) => {
+                use bigdecimal::{BigDecimal, FromPrimitive};
+
+                let lhs = BigDecimal::from_i128(*v1)?
```

Comment:
why not just use the infallible [`From<&i128>`](https://docs.rs/bigdecimal/latest/bigdecimal/struct.BigDecimal.html#impl-From%3C%26i128%3E-for-BigDecimal) directly?
```suggestion
                let lhs = BigDecimal::from(v1)?
```